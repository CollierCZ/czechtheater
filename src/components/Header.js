import React from 'react'
import PropTypes from 'prop-types'

import logo from '../images/logo.jpg'

const Header = (props) => (
    <header id="header" style={props.timeout ? {display: 'none'} : {}}>
        <div className="logo">
            <img src={logo} alt="" />
        </div>
        <div className="content">
            <div className="inner">
                <h1>Czech Theater</h1>
                <p>A multicultural amateur community theater in Brno presenting Czech plays in English.</p>
            </div>
        </div>
        <nav>
            <ul>
                <li><a href="javascript:;" onClick={() => {props.onOpenArticle('about')}}>About</a></li>
                <li><a href="javascript:;" onClick={() => {props.onOpenArticle('upcoming')}}>Future</a></li>
                <li><a href="javascript:;" onClick={() => {props.onOpenArticle('past')}}>Past</a></li>
                <li><a href="javascript:;" onClick={() => {props.onOpenArticle('contact')}}>Contact</a></li>
            </ul>
        </nav>
    </header>
)

Header.propTypes = {
    onOpenArticle: PropTypes.func,
    timeout: PropTypes.bool
}

export default Header
