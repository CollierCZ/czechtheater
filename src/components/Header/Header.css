header {
    margin: 0 auto;
    max-width: 960px;
    padding: 1.45rem 1.0875rem;
    margin-bottom: 1.45rem;
}

header img {
    margin: auto 0;
    width: 5.5rem;
    height: 5.5rem;
    border-radius: 100%;
}

header a {
    color: black;
    text-decoration: none;
    display: flex;
}

header .siteName {
    margin: auto 0.5em;
    font-size: 1.75em;
    font-weight: bold;
}