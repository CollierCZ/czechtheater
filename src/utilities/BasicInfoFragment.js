import { graphql } from 'gatsby';

export const BasicInfoFragment = graphql`
    fragment BasicInfoFragment on KenticoCloudItemBasicInfo {
        elements {
            about_us {
              value
            }
            contact_info {
              resolvedHtml
            }
            favicon {
                assets{
                  url
                }
            }
            name {
              value
            }
            short_description {
              value
            }
            social_media {
              value
            }
          }
    }
`;


