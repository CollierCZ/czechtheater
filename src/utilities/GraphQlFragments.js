import { graphql } from 'gatsby';

export const BasicInfoFragment = graphql`
    fragment BasicInfoFragment on KontentItemBasicInfo {
        elements {
            about_us {
              value
            }
            contact_info {
              resolvedData {
                html
              }
            }
            favicon {
              value {
                url
              }
            }
            name {
              value
            }
            short_description {
              value
            }
            social_media {
              value
            }
          }
    }
`;

export const ShowListFragment = graphql`
    fragment ShowListFragment on KontentItemShowSection {
      elements {
        shows {
          linked_items {
            elements {
              main_image {
                value {
                  url
                  description
                }
              }
              name {
                value
              }
              url {
                value
              }
              short_description {
                value
              }
              ticket_link {
                value
              }
              gallery {
                value {
                  url
                  description
                }
              }
            }
          }
        }
      }
    }
`;


