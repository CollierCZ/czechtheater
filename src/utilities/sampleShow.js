export const sampleShow = {
    elements: {
      name: {
        value: "Alquist's revenge"
      },
      description: {
        resolvedHtml: "All the acting"
      },
      main_image: {
        assets: [{
          description: "A cool image",
          url: "https://example.com"
        }]
      },
      ticket_link: {
        value: "https://example.com"
      },
      url: {
        value: "alquists-revenge"
      },
      gallery: {
          assets: [
              {
                  description: "A cool image",
                  url: "https://example.com"
              }
          ]
      }
    }
  }