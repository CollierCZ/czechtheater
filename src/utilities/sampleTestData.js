export const sampleImage = {
  description: "A great image",
  url: "https://example.com/wide.png"
};

export const sampleTallImageData = {
  description: "A great image",
  height: 250,
  width: 150,
  url: "https://example.com/wide.png"
};

export const sampleWideImageData = {
  description: "A great image",
  height: 150,
  width: 250,
  url: "https://example.com/wide.png"
};


export const sampleShow = {
  elements: {
    name: {
      value: "Alquist's revenge",
    },
    description: {
      value: "<p>All the acting in a longer way.</p>",
    },
    main_image: {
      value: [
        sampleImage
      ],
    },
    short_description: {
      value: "<p>All the acting</p>",
    },
    ticket_link: {
      value: "https://example.com",
    },
    url: {
      value: "alquists-revenge",
    },
    gallery: {
      value: [
        sampleImage
      ],
    },
  },
};

let sampleShowForPage = JSON.parse(JSON.stringify(sampleShow));
sampleShowForPage.elements.main_image.value = [
  sampleImage,
];

let sampleShowNoTicketLink = JSON.parse(JSON.stringify(sampleShow));
sampleShowNoTicketLink.elements.ticket_link.value = "";

let sampleShowForPageNoGallery = JSON.parse(JSON.stringify(sampleShowForPage));
sampleShowForPageNoGallery.elements.gallery.value = [];

let sampleShowForPageNoTicketLink = JSON.parse(
  JSON.stringify(sampleShowForPage)
);
sampleShowForPageNoTicketLink.elements.ticket_link.value = "";

export {
  sampleShowForPage,
  sampleShowForPageNoGallery,
  sampleShowForPageNoTicketLink,
  sampleShowNoTicketLink,
};

export const basicInfoQuery = {
  kontentItemBasicInfo: {
    elements: {
      name: {
        value: "Czech Theater",
      },
      favicon: {
        value: [
          {
            description: "Czech Theater",
            url: "https://assets-us-01.kc-usercontent.com/fb599585-baec-0077-1624-981d3321fff0/4e43fc7c-ce8d-49fd-aba9-bdf908330220/logo.jpg",
          },
        ],
      },
      short_description: {
        value: "A great theater.",
      },
    },
  },
};
