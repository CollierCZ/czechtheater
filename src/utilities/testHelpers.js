import { shallow } from "enzyme"

export const simpleShallowRender = (component) => {
  const wrapper = shallow (component)
  expect(wrapper).toMatchSnapshot()
}