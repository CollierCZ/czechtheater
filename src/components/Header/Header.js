import { Link } from "gatsby";
import PropTypes from "prop-types";
import React from "react";
import { ImageElement } from "@kontent-ai/gatsby-components";
import "./Header.css";

const Header = ({ image, siteTitle }) => (
  <header>
    <Link to="/">
      <ImageElement
        image={image}
        alt={image.description || ""}
        layout="fixed"
        height={100}
        width={100}
      />
      <div className="siteName">{siteTitle}</div>
    </Link>
  </header>
);

Header.propTypes = {
  siteTitle: PropTypes.string
};

Header.defaultProps = {
  siteTitle: ``
};

export default Header;
