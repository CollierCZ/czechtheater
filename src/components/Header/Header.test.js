import React from "react";
import Header from "./Header";

import { sampleImage } from "../../utilities/sampleTestData";
import { simpleShallowRender } from "../../utilities/testHelpers"

describe("Header", () => {
  it("loads correctly", () => {
    simpleShallowRender(<Header image={sampleImage} siteTitle="Czech Theater" />)
  })
})