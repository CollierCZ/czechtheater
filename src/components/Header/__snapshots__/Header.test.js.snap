// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header loads correctly 1`] = `
<header>
  <mockConstructor
    to="/"
  >
    <Memo(ImageElement)
      alt="A great image"
      height={100}
      image={
        Object {
          "description": "A great image",
          "url": "https://example.com/wide.png",
        }
      }
      layout="fixed"
      width={100}
    />
    <div
      className="siteName"
    >
      Czech Theater
    </div>
  </mockConstructor>
</header>
`;
